### ISSUE TITLE
unexpected url: file://index.android.bundle

### ISSUE THREAD

**Issue by awesomejerry** [facebook/react-native#7998]
I've assembled the release apk and install it to my device, but it crashes.
I'm using rn@0.26.1

Here's the stacktrace from my crash report
```
Fatal Exception: java.lang.IllegalArgumentException: unexpected url: file://index.android.bundle
       at com.squareup.okhttp.Request$Builder.url(Request.java:163)
       at com.facebook.react.modules.network.NetworkingModule.sendRequest(NetworkingModule.java:159)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.facebook.react.bridge.BaseJavaModule$JavaMethod.invoke(BaseJavaModule.java:316)
       at com.facebook.react.bridge.NativeModuleRegistry$ModuleDefinition.call(NativeModuleRegistry.java:187)
       at com.facebook.react.bridge.NativeModuleRegistry.call(NativeModuleRegistry.java:62)
       at com.facebook.react.bridge.CatalystInstanceImpl$NativeModulesReactCallback.call(CatalystInstanceImpl.java:432)
       at com.facebook.react.bridge.queue.NativeRunnableDeprecated.run(NativeRunnableDeprecated.java)
       at android.os.Handler.handleCallback(Handler.java:808)
       at android.os.Handler.dispatchMessage(Handler.java:103)
       at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
       at android.os.Looper.loop(Looper.java:193)
       at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:187)
       at java.lang.Thread.run(Thread.java:841)
```

**Comment by charpeni** [comment]
@facebook-github-bot label Android

**Comment by charpeni** [comment]
How did you build the release APK? 

You can check here how to do that : http://facebook.github.io/react-native/docs/signed-apk-android.html#generating-the-release-apk

**Comment by AwesomeJerry** [comment]
I was using "cd android && ./gradlew assembleRelease" to generate the apk. Everything works fine until it suddenly crashes all the time with the same weird error. Maybe it's a rare-happened case, seeking for anyone who encounters the same error.

**Comment by mohanrohith** [comment]
Any update on this issue? Getting the same error in between

**Comment by mohanrohith** [comment]
*bump*

**Comment by kfiroo** [comment]
@mohanrohith Hi, unfortunately your suggestion didn't work for me :(
I also tried to completely remove react-native-fabric-crashlytics from my project, that didn't work either.

any ideas?

**Comment by mohanrohith** [comment]
There should be some JS error happening on runtime like "variable undefined" which should be handled by some error handler.

Attaching a gist which I used.
https://gist.github.com/mohanrohith/2ba9655e29110ece390024adc4930fb1

And initialise in index.android.js  as `Errorutils.init()`

**Comment by jbrodriguez** [comment]
Hi @mohanrohith, I hit this issue too.

So your gist, ErrorUtils.js, would be a replacement for react-native-fabric-crashlytics ?

**Comment by mohanrohith** [comment]
@jbrodriguez Yes

**Comment by kfiroo** [comment]
@jbrodriguez @mohanrohith 
I think this error might not be related to `react-native-fabric-crashlytics`
I simply had some JS error in my project that caused the same error, solved it and it was gone without changing anything in `react-native-fabric-crashlytics`

**Comment by mohanrohith** [comment]
@kfiroo Yes, JS error in our project causes this issue. But this specific error message comes when that JS error is being handled by the errorhandler in `react-native-fabric-crashlytics`  which internally makes  a network request to get to the sourcecode. And the url to the source code is `file://` not `http://` which is not supported in okhttp library.

**Comment by kfiroo** [comment]
@mohanrohith Oh, I get it now :)
Thanks!

Did we create a PR for `react-native-fabric-crashlytics` to fix this yet?

Anyone wants to do it? or I can do it.

**Comment by kfiroo** [comment]
@mohanrohith looks like it is fixed in a new version 0.1.6 :)
Yay.

**Comment by mkonicek** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with ProductPains! Check out its new home: https://productpains.com/post/react-native/unexpected-url-fileindexandroidbundle

Product Pains has been very useful in highlighting the top bugs and feature requests:
https://productpains.com/product/react-native?tab=top

Also, if this issue is a bug, **please consider sending a pull request with a fix**.
